### PR TITLE
Add strict analysis modes

### PR DESCRIPTION
**Pull request by parlough**

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -1,29 +1,12 @@
...
-#
-# The issues identified by the analyzer are surfaced in the UI of Dart-enabled
-# IDEs (https://dart.dev/tools#ides-and-editors). The analyzer can also be
-# invoked from the command line by running `flutter analyze`.
-
-# The following line activates a set of recommended lints for Flutter apps,
-# packages, and plugins designed to encourage good coding practices.
 include: package:flutter_lints/flutter.yaml
 
+analyzer:
```
Comment (devoncarew) [inline analysis_options.yaml:3]: 👍

**Review by devoncarew — APPROVED** [review]
Thanks :)

For FYI, this is strictly an un-official investigative spike - what would dartpad look like w/ slightly fewer features and configurations and w/ a flutter web UI.

**Comment by parlough** [comment]
I assumed so, but I'm excited for the exploration. This is the time to try.

Beyond that, I'm hoping to introduce non-DartPad embed runnable code samples on dart.dev and a simpler reference user of dart-services will be also useful for me :)

**Comment by devoncarew** [comment]
> I assumed so, but I'm excited for the exploration. This is the time to try.
> 
> Beyond that, I'm hoping to introduce non-DartPad embed runnable code samples on dart.dev and a simpler reference user of dart-services will also be useful for me. Maybe there is some parts that can be split out and shared. :)

Interesting! I'm curious how you'd scope that use case; remove edibility? Just render well in a smaller UI?

**Comment by parlough** [comment]
It's part of a bunch of code snippet/sample work I am still planning as it is lower priority then a bunch of other work, but a whole DartPad iframe is not necessary in most cases. Especially in language documentation, where there are many samples, but we couldn't use them much due to the increases in load time and often poor user experience. 

I'm imagining edibility isn't needed much in more reference style documentation, and uses cases where it is, such as in learning experiences like tutorials, codelabs, workshops, etc DartPad or an equivalent can still be used. 

I'm hoping it could potentially be used in generated dart doc documentation as well.

I plan to work on a proper design doc, hopefully in July, and I can make sure to share that with you!

**Comment by devoncarew** [comment]
> I plan to work on a proper design doc for the devrel team, hopefully in July, and I can make sure to share that with you!

Sounds great! There are some people who've worked on the doc samples on api.flutter.dev who would be good to loop in at that time (familiarity with the use case, ideas for how they might consume a lighter-weight embedding, ...).

**Pull request merged**